src/spec.c

```diff
@@ -604,6 +595,34 @@ void IndexSpec_Free(void *ctx) {
   rm_free(spec);
 }
 
+static void *IndexSpec_FreeAsync(void *data) {
+  IndexSpec *spec = data;
+  RedisModuleCtx *threadCtx = RedisModule_GetThreadSafeContext(NULL);
+  RedisSearchCtx sctx = SEARCH_CTX_STATIC(threadCtx, spec);
+  RedisModule_AutoMemory(threadCtx);
+  RedisModule_ThreadSafeContextLock(threadCtx);
+
+  Redis_DropIndex(&sctx, true, false);
+  IndexSpec_FreeInternals(spec);
+
+  RedisModule_ThreadSafeContextUnlock(threadCtx);
+  RedisModule_FreeThreadSafeContext(threadCtx);
+  return NULL;
+}
+
+void IndexSpec_Free(void *ctx) {
+
+  IndexSpec *spec = ctx;
+
+  if (spec->timeout != -1) {
```

Comment:
->temporary... maybe other flags for this? maybe we should use a function to hide?